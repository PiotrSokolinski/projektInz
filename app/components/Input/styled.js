import styled from 'styled-components'
import { Colors } from '../../themes/index'

export const Input = styled.input`
  box-shadow: none;
  width: 100%;
  height: 50px;
  font-weight: normal;
  text-indent: 15px;
  font-size: 15px;
  line-height: 140%;
  letter-spacing: 0.02em;
  border: 1px solid ${Colors.white};
  box-sizing: border-box;
  outline: none;
`

export const Image = styled.img`
  left: -18px;
  top: -34px;
  float: right;
  position: relative;
`
