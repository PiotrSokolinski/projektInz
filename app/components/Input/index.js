/**
 *
 * Input
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import { Field, Control, Help } from 'react-bulma-components/lib/components/form'
import * as Styled from './styled'
import Label from '../Label'

const Input = ({ error, id, label, imageProps, inputProps }) => (
  <Field id={id}>
    <Label error={error} label={label} />
    <Control>
      <Styled.Input {...inputProps} />
      {imageProps && <Styled.Image {...imageProps} />}
    </Control>
    {error && <Help color="danger">{error}</Help>}
  </Field>
)

Input.propTypes = {
  error: PropTypes.string,
  id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  inputProps: PropTypes.shape({}),
  label: PropTypes.string,
  width: PropTypes.string,
  imageProps: PropTypes.object,
}

Input.defaultProps = {
  inputProps: {
    onChange: () => {},
  },
  error: '',
  label: '',
  imageProps: null,
}

export default Input
