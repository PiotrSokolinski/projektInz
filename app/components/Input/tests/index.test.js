/**
 *
 * Tests for Input
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { IntlProvider } from 'react-intl'
import Input from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<Input />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Input label="Label" />
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render with error and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Input error="error" label="Label" />
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render without error and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Input error={null} label="Label" />
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })

  it('Should invoke onChange', () => {
    const mockOnChange = jest.fn()
    const { getByPlaceholderText } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Input error={null} inputProps={{ placeholder: 'Placeholder', onChange: mockOnChange }} label="Label" />
      </IntlProvider>,
    )
    const input = getByPlaceholderText('Placeholder')
    expect(input.textContent).toBe('')
    fireEvent.change(input, { target: { value: 'change' } })
    expect(mockOnChange).toHaveBeenCalledTimes(1)
  })
})
