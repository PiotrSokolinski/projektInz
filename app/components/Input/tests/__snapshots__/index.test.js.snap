// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should render with error and match the snapshot 1`] = `
<div
  class="field"
>
  <label
    class="styled__StyledLabel-oaisvy-0 gsmrPq"
  >
    Label
  </label>
  <div
    class="control"
  >
    <input
      class="styled__Input-ms4xi4-0 cMqlpT"
    />
  </div>
  <p
    class="help is-danger"
  >
    error
  </p>
</div>
`;

exports[`<Input /> Should render without error and match the snapshot 1`] = `
<div
  class="field"
>
  <label
    class="styled__StyledLabel-oaisvy-0 gJTorn"
  >
    Label
  </label>
  <div
    class="control"
  >
    <input
      class="styled__Input-ms4xi4-0 cMqlpT"
    />
  </div>
</div>
`;
